src/io/flutter/utils/EvalOnDartLibrary.java

```diff
@@ -0,0 +1,195 @@
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Invoke methods on a Dart library using the observatory protocol.
+ */
+public class EvalOnDartLibrary implements Disposable {
+  private final IsolatesInfo.IsolateInfo isolateInfo;
+  private final VmService vmService;
+  private Isolate isolate;
+  private final String libraryName;
+  final CompletableFuture<LibraryRef> libraryRef;
+  private final Alarm myRequestsScheduler;
+
+  public EvalOnDartLibrary(String libraryName, FlutterDebugProcess debugProcess, VmService vmService) {
+    this.vmService = vmService;
+    final Collection<IsolatesInfo.IsolateInfo> isolates = debugProcess.getIsolateInfos();
+    assert(isolates.size() == 1);
+    isolateInfo = Iterables.get(isolates, 0);
+    this.libraryName = libraryName;
+    this.myRequestsScheduler = new Alarm(Alarm.ThreadToUse.POOLED_THREAD, this);
+    libraryRef = new CompletableFuture<>();
+    initialize();
+  }
+
+  public void dispose() {
+    myRequestsScheduler.dispose();
+    // TODO(jacobr): complete all pending futures as cancelled?
+  }
+
+  /**
+   * Workaround until the version of the VmService 'eval' method supports
+   * the scope argument.
+   */
+  private void callVmServiceRequest(VmService vmService, String methodName, JsonObject params, EvaluateConsumer consumer) {
+    try {
+      final Class<?> c = Class.forName("org.dartlang.vm.service.VmServiceBase");
+
+      final Method[] allMethods = c.getDeclaredMethods();
+      Method method = null;
+      for (Method m : allMethods) {
+        String mname = m.getName();
+        if (mname.equals("request")) {
+          method = m;
+          break;
+        }
+      }
+
+      if (method == null) {
+        throw new RuntimeException("Cannot find method 'Request'");
+      }
+      try {
+        method.setAccessible(true);
+        method.invoke(vmService, methodName, params, consumer);
+        // Handle any exceptions thrown by method to be invoked.
+      } catch (InvocationTargetException x) {
+        final Throwable cause = x.getCause();
+        java.lang.System.err.format("invocation of request failed: %s%n",
+                                    methodName, cause.getMessage());
+      }
+    } catch (IllegalAccessException|ClassNotFoundException x) {
+      throw new RuntimeException((x.toString()));
+    }
+  }
+
+  public CompletableFuture<InstanceRef> eval(String expression, Map<String, String> scope) {
+    final CompletableFuture<InstanceRef> future = new CompletableFuture<>();
+    myRequestsScheduler.addRequest(() -> {
+      libraryRef.thenAcceptAsync((LibraryRef ref) -> {
+        evaluateHelper(isolateInfo.getIsolateId(), ref.getId(), expression, scope,
+         new EvaluateConsumer() {
+           @Override
+           public void onError(RPCError error) {
+             // TODO(jacobr): what is the right way to log this case?
+             System.err.println("Got RPCError: " + error);
```

Comment:
Presumably the client will listen to the error result and display (or ignore) appropriately to the user? If you do need to log, you can use the intellij logging system: https://github.com/flutter/flutter-intellij/blob/335e25cc3c97d515305ec72ec4dd48dd3ebba8e8/src/io/flutter/run/daemon/DaemonEvent.java#L290

Reply:
Done